wandb/sdk/launch/sweeps/scheduler_sweep.py

```diff
@@ -38,120 +37,114 @@ class SweepScheduler(Scheduler):
-            if not self.is_alive():
-                return
-            # AgentHeartbeat wants dict of runs which are running or queued
-            _run_states = {}
-            for run_id, run in self._yield_runs():
-                if run.state == SimpleRunState.ALIVE:
-                    _run_states[run_id] = True
-            _msg = (
-                f"{LOG_PREFIX}AgentHeartbeat sending: \n{pprint.pformat(_run_states)}\n"
-            )
-            logger.debug(_msg)
-            # TODO(hupo): Should be sub-set of _run_states specific to worker thread
-            commands = self._api.agent_heartbeat(
-                self._heartbeat_agents[worker_idx].id, {}, _run_states
-            )
-            if commands:
-                _msg = f"{LOG_PREFIX}AgentHeartbeat received {len(commands)} commands: \n{pprint.pformat(commands)}\n"
-                logger.debug(_msg)
-                for command in commands:
-                    _type = command.get("type")
-                    # type can be one of "run", "resume", "stop", "exit"
-                    if _type == "exit":
-                        self.state = SchedulerState.COMPLETED
-                        self.exit()
-                        return
-                    if _type == "stop":
-                        # TODO(hupo): Debug edge cases while stopping with active runs
-                        self.state = SchedulerState.COMPLETED
-                        self.exit()
-                        return
-                    run = SweepRun(
-                        id=command.get("run_id"),
-                        args=command.get("args"),
-                        logs=command.get("logs"),
-                        program=command.get("program"),
-                    )
-                    with self._threading_lock:
+    def _heartbeat(self, worker_id: int) -> None:
+        # Make sure Scheduler is alive
+        if not self.is_alive():
+            return
+        # AgentHeartbeat wants a Dict of runs which are running or queued
+        _run_states: Dict[str, bool] = {}
+        for run_id, run in self._yield_runs():
+            # Filter out runs that are from a different worker thread
+            if run.worker_id == worker_id and run.state == SimpleRunState.ALIVE:
+                _run_states[run_id] = True
+        logger.debug(
+            f"{LOG_PREFIX}AgentHeartbeat sending: \n{pprint.pformat(_run_states)}\n"
+        )
+        commands: List[Dict[str, Any]] = self._api.agent_heartbeat(
+            self._workers[worker_id].agent_id,  # agent_id: str
+            {},  # metrics: dict
+            _run_states,  # run_states: dict
+        )
+        logger.debug(
+            f"{LOG_PREFIX}AgentHeartbeat received {len(commands)} commands: \n{pprint.pformat(commands)}\n"
+        )
+        if commands:
+            for command in commands:
+                # The command "type" can be one of "run", "resume", "stop", "exit"
+                _type = command.get("type", None)
```

Comment:
Is this better style than just `command.get("type")` ? Purely curious

Reply:
Probably a wash. It is more readable/obvious at the cost of extra verbosity.